dts/bindings/gpio/nxp,pcal6408a.yaml

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: PCAL6408A 8-bit I2C-based I/O expander
+
+compatible: "nxp,pcal6408a"
+
+include: [i2c-device.yaml, gpio-controller.yaml]
+
+properties:
+    label:
+      required: true
+
+    ngpios:
+      required: true
+      const: 8
+
+    int-gpios:
+      type: phandle-array
+      required: false
+      description: |
+        GPIO connected to the controller INT pin. This pin is active-low.
+
+    reset-gpios:
+      type: phandle-array
+      required: false
+      description: |
+        GPIO connected to the controller RESET pin. This pin is active-low.
+
+    init-out-low:
+      type: int
+      required: false
+      default: 0
+      description: |
+        Bit mask identifying pins that should be initialized as outputs
+        driven low.
+
+    init-out-high:
+      type: int
+      required: false
+      default: 0
+      description: |
+        Bit mask identifying pins that should be initialized as outputs
+        driven high.
```

Comment:
We want to base our DT on the Linux implementation. This functionality is implemented there using GPIO hogging mechanism as described in [GPIO device tree bindings](https://github.com/torvalds/linux/blob/7a7fd0de4a9804299793e564a555a49c1fc924cb/Documentation/devicetree/bindings/gpio/gpio.txt#L185) document. We haven't considered yet how this could be implemented in Zephyr. The above approach is however non-standard.

Reply:
I copied these two properties from the ["semtech,sx1509b" binding](https://github.com/zephyrproject-rtos/zephyr/blob/48321ef2525c14c4d9b6040afe8ae72ff07a8d1e/dts/bindings/gpio/semtech%2Csx1509b-gpio.yaml#L28-L42), but I actually don't use this functionality, so I'm happy to drop it. It can be added later if needed.

Reply:
This functionality is generally useful but I propose to drop it for now and implement later using GPIO hogging mechanism.